lib/jdomodels/src/main/java/org/sagebionetworks/repo/model/dbo/file/MultipartUploadComposerDAOImpl.java

```diff
@@ -0,0 +1,172 @@
+package org.sagebionetworks.repo.model.dbo.file;
+
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_COMPOSER_PART_RANGE_LOWER_BOUND;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_COMPOSER_PART_RANGE_UPPER_BOUND;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_COMPOSER_PART_UPLOAD_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_FILE_HANDLE_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_STATE;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_UPLOAD_ETAG;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.COL_MULTIPART_UPLOAD_ID;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_MULTIPART_UPLOAD;
+import static org.sagebionetworks.repo.model.query.jdo.SqlConstants.TABLE_MULTIPART_UPLOAD_COMPOSER_PART_STATE;
+
+import java.util.List;
+import java.util.UUID;
+
+import org.sagebionetworks.repo.model.dbo.DBOBasicDao;
+import org.sagebionetworks.repo.model.file.MultipartUploadState;
+import org.sagebionetworks.repo.model.upload.PartRange;
+import org.sagebionetworks.util.ValidateArgument;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.jdbc.core.RowMapper;
+import org.springframework.jdbc.core.namedparam.MapSqlParameterSource;
+import org.springframework.jdbc.core.namedparam.NamedParameterJdbcTemplate;
+
+public class MultipartUploadComposerDAOImpl implements MultipartUploadComposerDAO {
+
+	private static final String PARAM_UPLOAD_ID = "uploadId";
+	private static final String PARAM_FILE_HANDLE_ID = "fhId";
+	private static final String PARAM_ETAG = "etag";
+	private static final String PARAM_MULTIPART_STATE = "multipartState";
+	private static final String PARAM_LOWER_BOUND = "lowerBound";
+	private static final String PARAM_UPPER_BOUND = "upperBound";
+
+	private static final String SQL_SELECT_PARTS_BY_UPLOAD_ID =
+			"SELECT * FROM " + TABLE_MULTIPART_UPLOAD_COMPOSER_PART_STATE
+			+ " WHERE " + COL_MULTIPART_COMPOSER_PART_UPLOAD_ID + " = :" + PARAM_UPLOAD_ID;
+
+	private static final String SQL_SELECT_PARTS_IN_RANGE =
+			"SELECT "
+			+ COL_MULTIPART_COMPOSER_PART_RANGE_LOWER_BOUND + ", "
+			+ COL_MULTIPART_COMPOSER_PART_RANGE_UPPER_BOUND
+			+ " FROM " + TABLE_MULTIPART_UPLOAD_COMPOSER_PART_STATE
+			+ " WHERE " + COL_MULTIPART_COMPOSER_PART_UPLOAD_ID + " = :" + PARAM_UPLOAD_ID
+			+ " AND " + COL_MULTIPART_COMPOSER_PART_RANGE_LOWER_BOUND + " >= :" + PARAM_LOWER_BOUND
+			+ " AND " + COL_MULTIPART_COMPOSER_PART_RANGE_UPPER_BOUND + " <= :" + PARAM_UPPER_BOUND
+			+ " ORDER BY " + COL_MULTIPART_COMPOSER_PART_RANGE_LOWER_BOUND + " ASC";
+
+	private static final String FOR_UPDATE = " FOR UPDATE";

```

Comment:
Remove since the eTag is being updated, which grants an exclusive lock on the parent row with Read Committed